### PR TITLE
Feature/youtube start time

### PR DESCRIPTION
**Pull request by tterb**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Added support for start-times in youtube liquid-tags, using the syntax suggested in the feature request: `{% youtube EauykEv_2iA?t=15m26s %}`  
I'm fairly new to Rails, so any constructive feedback would be much appreciated 👍

## Related Tickets & Documents
Feature Request #107 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
None

## Added to documentation?
  - [ ] docs.dev.to
  - [ ] readme
  - [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
alt-text

### CODE REVIEW

File: app/liquid_tags/youtube_tag.rb [hunk 1]
```diff
@@ -34,8 +35,21 @@ def translate_url(input)
...
+      "#{id.split('?t=')[0]}?start=#{seconds}"
+    end
+  end
+
   def valid_id?(id)
-    id.length == 11 && id =~ /[a-zA-Z0-9_-]{11}/
+    if id.length == 11
+      id =~ /[a-zA-Z0-9_-]{11}/
+    elsif id.split("?")[0].length == 11
+      id =~ /[a-zA-Z0-9_-]{11}\?start\=\d*/
```
Comment (AntonyGarand) [inline app/liquid_tags/youtube_tag.rb]: You can make the `?start=\d` part optional with `?` and only need one regex:
```ruby
id =~ /[a-zA-Z0-9_-]{11}[\?start\=\d*]?/ 
```

Comment (tterb) [reply app/liquid_tags/youtube_tag.rb]: @AntonyGarand That's what I originally had tried to do in in 03674c479a2452e8f3111a69dd666017288dccb5, but it wasn't failing to catch the invalid youtube ID `really_long_invalid_id` and I couldn't find a resolution.  
This is the regex I was using:
```ruby
id =~ /[a-zA-Z0-9_-]{11}(?:\?start\=\d*)?/
``` 
The only difference being the `?:` to match the optional portion without making it a capturing group, but it's possible that this doesn't work as I expected in Ruby regex.

Comment (benhalpern) [reply app/liquid_tags/youtube_tag.rb]: I'm happy to go with this implementation, as the logic is pretty clear. Even if more elegant, I can't imagine the pure regex solution being _easier_ to read or meaningfully cleaner.

@maestromac wanna give this all a once-over and merge if you feel good with it.

Comment (maestromac) [reply app/liquid_tags/youtube_tag.rb]: Hey @tterb. I took a look at it and I think our `really_long_invalid_id` spec is testing something that doesn't need to be tested. After all, if an id is invalid, youtube embed will let the user know there's a mistake. Feel free to remove that spec and apply either of those 1 line regex.

Comment (tterb) [reply app/liquid_tags/youtube_tag.rb]: @maestromac Okay, will do.

**Pull request merged**